core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java

```diff
@@ -1015,6 +1016,18 @@ default String[] getConsumes(final AnnotationScannerContext context) {
         return currentConsumes;
     }
 
+    default String[] getConsumesForRequestBody(final AnnotationScannerContext context) {
+        String[] currentConsumes = context.getCurrentConsumes();
+        if (currentConsumes == null || currentConsumes.length == 0) {
+            currentConsumes = context.getDefaultConsumes();
+        }
+        return currentConsumes;
+    }
+
+    default boolean isRequestBody(MethodParameterInfo mip) {
+        return mip.annotations().isEmpty();
```

Comment:
This could be a problem for cases like a request body annotated with the `@RequestBody`, `@Valid`, etc. I think the Spring override may also support `@Valid` and possibly other or custom annotations.

It may be safer to do something like what the request body scanner is doing: 
https://github.com/smallrye/smallrye-open-api/blob/6d4eecfa2133fd8741f79e488bd33b332998a8e8/core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java#L969-L972

Of course it won't be exactly the same unless the processing order is changed and the request body scanning is first.

Reply:
`@RequestBody` is JaxRS right ? I think the JaxRS scanner override that default method

Reply:
That's the MP OpenAPI one. I think there could be some number of annotations on a request body though.

Reply:
Ok, I am with you, some might be better to exclude the others (If NOT `@PathParam`, `@QueryParam` etc ?